makefile.shade

```diff
@@ -15,13 +17,25 @@ var AUTHORS='Microsoft Open Technologies, Inc.'
 use-standard-lifecycle
 k-standard-goals
 
-#repack-x86 target='compile' if='Directory.Exists("src")'
+default CONFIGURATION_LOCAL='${E("Configuration")}'
+
+#pack-test-dependencies target='repo-initialize'
+    rimraf rimrafDir='${TEST_ARTIFACTS_INITIAL_PACKAGE_DIR}' if='Directory.Exists(TEST_ARTIFACTS_INITIAL_PACKAGE_DIR)'
+
     @{
-        if (string.IsNullOrEmpty(CONFIGURATION_LOCAL))
-        {
-            CONFIGURATION_LOCAL = "Debug";
-        }
+        Directory.CreateDirectory(TEST_ARTIFACTS_INITIAL_PACKAGE_DIR);
+        var projectDirectory = Path.Combine(BASE_DIR_LOCAL, "src", TEST_PROJECT_PACKAGE_NAME);
         
+        Dotnet("restore " + projectDirectory);
+        Dotnet("pack " + projectDirectory + " -o " + TEST_ARTIFACTS_INITIAL_PACKAGE_DIR);
+
+        // Restore testapps folder locally to ensure all dynamically generated packages have been cached. 
+        var testProjectDirectory = Path.Combine(BASE_DIR_LOCAL, "testapps");
```

Comment:
Would this work when `repo-initialize` does a restore (does it still find the packages in that case)?